share/wake/lib/core/examples.wake

```diff
@@ -0,0 +1,134 @@
+
+# Simplify github json log
+export def example4 (log: JValue): Result JValue Error =
+    def simpleCommit commit =
+        require Pass message =
+            commit
+            | jField "commit" |> jField "message"
+            | expectOne ".commit.message missing"
+
+        require Pass name =
+            commit
+            | jField "commit" |> jField "name"
+            | expectOne ".commit.name missing"
+
+        JObject ("message" :-> message, "name" :-> name, Nil) | Pass
+
+    # This is a bit complicated IMO but not too bad
+    log
+    | jFlatten
+    | findFailFn simpleCommit
+    | rmap JArray
+
+# Simplify github json log
+export def example5 (log: JValue): Result JValue Error =
+    def simpleCommit commit =
+        require Pass message =
+            commit
+            | jField "commit" |> jField "message"
+            | expectOne ".commit.message missing"
+
+        require Pass name =
+            commit
+            | jField "commit" |> jField "name"
+            | expectOne ".commit.name missing"
+
+        def parents = commit | jField "parents" |> jFlatten |> jField "html_url" | JArray
+
+        JObject ("message" :-> message, "name" :-> name, "parents" :-> parents, Nil)
+        | Pass
+
+    # This is a bit complicated IMO but not too bad
+    log
+    | jFlatten
+    | findFailFn simpleCommit
+    | rmap JArray
+
+# Convert a git log to a simplified wake tuple
+export tuple Commit =
+    Message: String
+    Name: String
+    Parents: List String
+
+# Simplify github json log
+export def example6 (log: JValue): Result (List Commit) Error =
+    def simpleCommit commit =
+        require Pass message =
+            commit
+            | jField "commit" |> jField "message"
+            |> jString
+            | expectOne ".commit.message was missing or not a string"
+
+        require Pass name =
+            commit
+            | jField "commit" |> jField "name"
+            |> jString
+            | expectOne ".commit.name missing or not a string"
+
+        def parents = commit | jField "parents" |> jFlatten |> jField "html_url" |> jString
+
+        Commit message name parents
+        | Pass
+
+    # This is a bit complicated IMO but not too bad
+    |> log
+    |> jFlatten
+    | findFailFn simpleCommit
+
+export def example7 (json: JValue): List JValue =
+    |> json
+    |> jField "instances"
+    |  filter (_ | jField "instance_name" | jExists "testIndicator".JString)
+
+# It turns out, edit functions compose with `$` in the order you want
+# which is somehow surprising and obvious at the same time. Who knew?
+export def exampleEdit1: Pair Integer b => Pair Integer b =
+    editPairFirst $ (_+4)
+
+export def exampleEdit2: Triple (Pair a Integer) b c => Triple (Pair a Integer) b c =
+    editTripleFirst $ editPairSecond $ (_+4)
+
+export def exampleEdit3: JValue => JValue =
+    editJField "hello"
+    $ editJField "bar"
+    $ editJString
+    $ ("zzz{_}zzz")
```

Comment:
Do we care about end-of-file newlines?  The way my VS Code is configured, the next change I make to any of these files will wind up adding a bit of line noise to the PR.